src/display/annotation_layer.js

```diff
@@ -499,6 +499,7 @@ class TextWidgetAnnotationElement extends WidgetAnnotationElement {
     if (this.data.textAlignment !== null) {
       element.style.textAlign = TEXT_ALIGNMENT[this.data.textAlignment];
     }
+    storage.modified = false;
```

Comment:
Instead of tracking this at the call sites, which is error-prone since it may be forgotten, we can simply track this in the `AnnotationStorage` class itself internally, i.e., mark it as modified when `getOrCreate/setValue` is called with a non-existent value.

Reply:
This means we end up `modified` when we set the default values in the acroform.
@brendandahl suggested adding something like an `isInitialValue` kwarg to the set methods. I'll need to be able to reset the `modified` flag on save, so I'll add a method to allow for that.

Reply:
Done and done, but resetting the modified flag on save doesn't appear to be working.

Reply:
Fixed with https://github.com/mozilla/pdf.js/pull/12241/commits/8f17a49ba7848df5abd0ff4d6e8044c86d5b2bdc . I may be clearing `modified` slightly early. But we appear to be passing in the *values* of `annotationStorage` rather than the object; I could potentially change that.

Reply:
Resolved in https://github.com/mozilla/pdf.js/pull/12241/commits/3173e7235e37795e7e1d0e4386797f65a19dda1f .